dbms/src/TestUtils/executorSerializer.cpp

```diff
@@ -0,0 +1,295 @@
+{
+    context.buf.fmtAppend("{}|columns:{{", executor_id);
+    context.buf.joinStr(
+        proj.exprs().begin(),
+        proj.exprs().end(),
+        [&](const auto & expr, FmtBuffer &) {
+            serializeExpression(expr, context);
+        },
+        ", ");
+    context.buf.append("}\n");
+}
+
+void serializeAggregation(const String & executor_id, const tipb::Aggregation & agg, ExecutorSerializerContext & context)
+{
+    context.buf.fmtAppend("{}|group_by: columns:{{", executor_id);
+    context.buf.joinStr(
+        agg.group_by().begin(),
+        agg.group_by().end(),
+        [&](const auto & by_item, FmtBuffer &) {
+            serializeExpression(by_item, context);
+        },
+        ", ");
+    context.buf.append("}, agg_func:{");
+    context.buf.joinStr(
+        agg.agg_func().begin(),
+        agg.agg_func().end(),
+        [&](const auto & by_item, FmtBuffer &) {
+            serializeExpression(by_item, context);
+        },
+        ", ");
+    context.buf.append("}\n");
+}
+
+void serializeTopN(const String & executor_id, const tipb::TopN & top_n, ExecutorSerializerContext & context)
+{
+    context.buf.fmtAppend("{}|order_by: columns{{", executor_id);
+    context.buf.joinStr(
+        top_n.order_by().begin(),
+        top_n.order_by().end(),
+        [&](const auto & by_item, FmtBuffer & fb) {
+            serializeExpression(by_item.expr(), context);
+            fb.fmtAppend(", desc: {}", by_item.desc());
+        },
+        ", ");
+    context.buf.fmtAppend("}}, limit: {}\n", top_n.limit());
+}
+
+void serializeJoin(const String & executor_id, const tipb::Join & join, ExecutorSerializerContext & context)
+{
+    context.buf.fmtAppend("{}|{},{}. left_join_keys: {{", executor_id, getJoinTypeName(join.join_type()), getJoinExecTypeName(join.join_exec_type()));
+    context.buf.joinStr(
+        join.left_join_keys().begin(),
+        join.left_join_keys().end(),
+        [&](const auto & key, FmtBuffer & fb) {
+            fb.fmtAppend("type: {}", getFieldTypeName(key.field_type().tp()));
+        },
+        ", ");
+    context.buf.append("}, right_join_keys: {");
+    context.buf.joinStr(
+        join.left_join_keys().begin(),
+        join.left_join_keys().end(),
+        [&](const auto & key, FmtBuffer & fb) {
+            fb.fmtAppend("type: {}", getFieldTypeName(key.field_type().tp()));
+        },
+        ", ");
+    context.buf.append("}\n");
+}
+
+void serializeExchangeSender(const String & executor_id, const tipb::ExchangeSender & sender, ExecutorSerializerContext & context)
+{
+    context.buf.fmtAppend("{}|type:{}\n", executor_id, getExchangeTypeName(sender.tp()));
+}
+
+void serializeExchangeReceiver(const String & executor_id, const tipb::ExchangeReceiver & receiver, ExecutorSerializerContext & context)
+{
+    context.buf.fmtAppend("{}|type:{}\n", executor_id, getExchangeTypeName(receiver.tp()));
```

Comment:
add the schema of ExchangeReceiver?

Reply:
I will modify receiver's type to match the type of sender too